pkg/kubelet/kubelet_pods.go

```diff
@@ -1020,6 +1021,69 @@ func (kl *Kubelet) removeOrphanedPodStatuses(pods []*v1.Pod, mirrorPods []*v1.Po
 	kl.statusManager.RemoveOrphanedStatuses(podUIDs)
 }
 
+// deleteMirrorPod uses mirror client to delete the mirror pod from the API server
+func (kl *Kubelet) deleteMirrorPod(podFullName string) {
+	deleted, err := kl.podManager.DeleteMirrorPod(podFullName, nil)
+	if err != nil {
+		klog.Errorf("encountered error when deleting mirror pod %q : %v", podFullName, err)
+		return
+	}
+	if !deleted {
+		klog.Errorf("mirror pod %q was not deleted", podFullName)
+		return
+	}
+	// the mirror pod has been removed from api server, clean the entry for bookkeeping
+	kl.podManager.MarkPodTerminated(podFullName)
+}
+
+// deleteOrphanedMirrorPods checks whether grace period for orphaned mirror pod has elapsed.
+// Once the grace period has passed, podManager.DeleteMirrorPod() is called to delete mirror pod
+// from the API server
+func (kl *Kubelet) deleteOrphanedMirrorPods() {
+	podFullNames := kl.podManager.GetOrphanedMirrorPodNames()
```

Comment:
note to self, at this point:

`SyncLoop (REMOVE, "file")` record would have removed the pod from the pod manager
mirror pod record remains in pod manager itself